core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -60,14 +61,37 @@
 
     static final String DISABLE_DYNAMIC_SCRIPTING_SETTING = "script.disable_dynamic";
 
+    // a parsing function that requires a non negative int and a timevalue as arguments split by a slash
+    // this allows you to easily define rates
+    static final Function<String, Tuple<Integer, TimeValue>> MAX_COMPILATION_RATE_FUNCTION =
+            (String value) -> {
+                if (value.contains("/") == false || value.startsWith("/") || value.endsWith("/")) {
+                    throw new IllegalArgumentException("parameter must contain a positive number and a timevalue, i.e. 10/1m");
+                }
+                int idx = value.indexOf("/");
+                String count = value.substring(0, idx);
+                String time = value.substring(idx + 1);
+
+                int rate = Integer.parseInt(count);
+                if (rate < 0) {
+                    throw new IllegalArgumentException("rate ["+ rate +"] must be positive");
+                }
+                TimeValue timeValue = TimeValue.parseTimeValue(time, "script.max_compilations_rate");
+                if (timeValue.nanos() <= 0) {
```

Comment:
I wonder if we should have a minimum time value. For example, if someone does a rate per 1s, this is something we are unlikely able to handle reliably.  Perhaps the minimum should be per 1 minute?

Reply:
++, set it to 1 minute